src/org/wso2/tg/jenkins/util/ConfigUtils.groovy

```diff
@@ -0,0 +1,165 @@
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.tg.jenkins.util
+
+/**
+ * Populates the TestGrid yaml input parameters in infra,deployment,scenario sections
+ * according to the config properties file definitions
+ *
+ * Ex :
+ * common.xxx=yyyy (insert the input parameter to all three sections)
+ * infra.default.xxx=yyy (insert to input parameter section of infra section by default)
+ * infra.xxx=yyy (insert to input parameter section of infra section only if user has specified in the
+ * testgridProvidedParameters section
+ *
+ * @param tgYamlContent parsed testgrid yaml file content
+ * @param commonConfigs parsed common-config  properties file content
+ * @return testgrid yaml file content with input parameters correctly set
+ */
+def addCommonConfigsToTestGridYaml(tgYamlContent, commonConfigs) {
+
+    echo "Populating testgrid yaml file with common configs"
+
+    commonConfigs.each { key, value ->
+        def infraProvisioners = null;
+        def deploymentPatterns = null;
+        def scenarioConfigs = null;
+
+        if (tgYamlContent.infrastructureConfig !=null ){
+            infraProvisioners = tgYamlContent.infrastructureConfig.provisioners
+        }
+        if(tgYamlContent.deploymentConfig!=null){
+            deploymentPatterns = tgYamlContent.deploymentConfig.deploymentPatterns
+        }
+
+        scenarioConfigs = tgYamlContent.scenarioConfigs
+
+        //split each property from . symbol
+        def split = key.split('\\.');
+
+        if (split[0].equals("common")) {
+            //apply to all  the available inputParams
+            addToInputParams(infraProvisioners, split[split.length - 1], value)
+            addToInputParams(deploymentPatterns, split[split.length - 1], value)
+            if(scenarioConfigs.inputParameters.get(0) != null ){
+                scenarioConfigs.inputParameters.get(0).put(split[split.length - 1], value)
+            }
+
+
+        } else if (split[0].equals("infra")) { //apply to only infraConfig section
+            //proceed if only valid entry
+            if (split.length >= 2 && infraProvisioners != null) {
+
+                //add to inputParams if default prefix is present
+                if (split[1].equals("default")) {
+                    addToInputParams(infraProvisioners, split[split.length - 1], value)
+                } else {
+                    //else only add if user has specified
```

Comment:
One software best practice we follow is that, if you have to add a line comment, then probably the subsequent code-segment need to go to its own method. :) 
As you know, the method names itself can be self-explanatory. We can add a javadoc if necessary on top of that.